pkg/datapath/linux/datapath.go

```diff
@@ -72,3 +77,19 @@ func (l *linuxDatapath) InstallProxyRules(proxyPort uint16, ingress bool, name s
 func (l *linuxDatapath) RemoveProxyRules(proxyPort uint16, ingress bool, name string) error {
 	return iptables.RemoveProxyRules(proxyPort, ingress, name)
 }
+
+func (l *linuxDatapath) WriteTemplateConfig(w io.Writer, e datapath.EndpointConfiguration) error {
```

Comment:
I guess `linuxDatapath` is a `ConfigWriter` since the `Datapath` interface is composed with the `ConfigWriter` interface. But, the crux of your question seems to be why does a `Datapath` have to be composed with a `ConfigWriter` - is that correct? 

The endpoint package uses the `Datapath` of its `Owner` to do these data path configuration updates, and I didn't want to change that up in this PR.

Let me know if I am on the right track - we can discuss offline as well, as I want to make sure I am grasping your question correctly.

Reply:
My question looking at this is why do we need to write all this boilerplate? Why not just publicly compose the inner ConfigWriter?

Reply:
can you provide an example of what you mean by "publicly compose"? e.g., `Datapath().ConfigWriter()`?

Reply:
Discussed offline with @joestringer  - `linuxDatapath` is now composed with a `ConfigWriter` . @raybejjani is this what you had in mind?